internal/vshard/cluster.go

```diff
@@ -373,10 +376,23 @@ func (c *Cluster) Discover() {
 	if c.snapshot.Created <= ns.Created {
 		ns.UpdatePriorities(c.snapshot.priorities)
 		c.snapshot = ns
+
+		go c.saveSnapshotToStorage(ns.Copy())
 	}
 	c.mutex.Unlock()
 }
 
+func (c *Cluster) saveSnapshotToStorage(ns Snapshot) {
```

Comment:
You do not know anything about what callback does, so just call in appropriate place:

```go
if c.onClusterDiscovered != nil {
  c.onClusterDiscovered()
}
```